### PR TITLE
use template needs to work also with starter template

### LINKED ISSUE #9256: `quarto use template` no longer works for directories without a `_extensions/` folder
### Bug description

`quarto use template` used to work to copy the contents on a Git repo, regardless of its contents. It now seems to fail if there is no `_extensions/` directory.

### Steps to reproduce

```
quarto use template EmilHvitfeldt/website-template
```

Follow prompts:
```
Quarto templates may execute code when documents are rendered. If you do not 
trust the authors of the template, we recommend that you do not install or 
use the template.
? Do you trust the authors of this template (Y/n) › Yes
? Create a subdirectory for template? (Y/n) › Yes
? Directory name: › website
```

### Expected behavior

In 1.4.551:
```
[✓] Downloading
[✓] Unzipping
[✓] Copying files...

Files created:
 - _quarto.yml
 - index.qmd
 - styles.scss
 - blog
 - about.qmd
 - _site
 - profile.png
 - about.jpg
 - blog.qmd
 ```

### Actual behavior

On `main`:
```
[✓] Downloading
[✓] Unzipping
ERROR: NotFound: No such file or directory (os error 2): readdir '/var/folders/1q/1yytldpd03z8426k84g_tzsc0000gp/T/quarto-session7d84bbb9865dbec1/d25b5c65803b11d5/ba8e12719bcb7bd3/archive/website-template-main/_extensions'

Stack trace:
    at Object.readDirSync (ext:deno_fs/30_fs.js:211:10)
    at readExtensions (file:///Users/charlottewickham/Documents/posit/quarto-cli/src/extension/extension.ts:416:30)
    at useTemplate (file:///Users/charlottewickham/Documents/posit/quarto-cli/src/command/use/commands/template.ts:88:38)
    at eventLoopTick (ext:core/01_core.js:153:7)
    at async Command.actionHandler (file:///Users/charlottewickham/Documents/posit/quarto-cli/src/command/use/commands/template.ts:52:7)
    at async Command.execute (file:///Users/charlottewickham/Documents/posit/quarto-cli/src/vendor/deno.land/x/cliffy@v1.0.0-rc.3/command/command.ts:1948:7)
    at async Command.parseCommand (file:///Users/charlottewickham/Documents/posit/quarto-cli/src/vendor/deno.land/x/cliffy@v1.0.0-rc.3/command/command.ts:1780:14)
    at async quarto (file:///Users/charlottewickham/Documents/posit/quarto-cli/src/quarto.ts:156:3)
    at async file:///Users/charlottewickham/Documents/posit/quarto-cli/src/quarto.ts:170:5
    at async mainRunner (file:///Users/charlottewickham/Documents/posit/quarto-cli/src/core/main.ts:35:5)
```

### Your environment

- Mac OS 14.4.1

### Quarto check output

```bash
Quarto 99.9.9
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.10.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 99.9.9
      commit: 26172cb90eb6c2d41742df662771c1d4b2197a3f
      Path: /Users/charlottewickham/Documents/posit/quarto-cli/package/dist/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.03.13
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/charlottewickham/Library/TinyTeX/bin/universal-darwin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.2
      Path: /Users/charlottewickham/.pyenv/versions/3.12.2/bin/python3
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.3
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
      knitr: 1.45
      rmarkdown: 2.26

[✓] Checking Knitr engine render......OK
```

### PR DESCRIPTION
**Pull request by cderv**
`quarto use template` can be used to copy files from a remote even if it does not have extension

Tweaks from #8764 regarding installation of extension only needs to happen if there is an `_extension` folder

#8764 uses 
https://github.com/quarto-dev/quarto-cli/blob/0ba1c27665b288e3b61f4dffd519b6c88769fc6e/src/command/use/commands/template.ts#L83-L104

This PR makes sure that `readExtensions` returns empty list when `extensionsDirectory` doesn't exists. This should prevent any similar issue when using `readExtensions` in the future. 

closes #9256 

I did not add to changelog because it was only broken by #8764 merged in 1.5.15

### CODE REVIEW

**Review by cscheid — APPROVED** [review]
Looks good. I searched for additional instances of using `extensionsDirectory` and this seemed to be the only one.

**Comment by cderv** [comment]
This would be `readExtensions()` usage maybe to look for. 

I am seeing some occurrences where we externally do the existence-checking
https://github.com/quarto-dev/quarto-cli/blob/0ba1c27665b288e3b61f4dffd519b6c88769fc6e/src/extension/extension.ts#L694-L697
https://github.com/quarto-dev/quarto-cli/blob/0ba1c27665b288e3b61f4dffd519b6c88769fc6e/src/extension/install.ts#L388-L393

Looking deeper we do have this helper 
https://github.com/quarto-dev/quarto-cli/blob/0ba1c27665b288e3b61f4dffd519b6c88769fc6e/src/extension/install.ts#L610-L624

But it was modified not to return `undefined` (53ef1d97c1689c211dfd86d2f5290078c3bf9e56), which seems odd to me as we still checked for it in some places and it makes more sense 🤔 

Anyhow, I am thinking now that this fix should probably be more scoped to `template.ts` by just reading if the dir exists :

**Comment by cderv** [comment]
> Or I should check all call to readExtensions()

I went with that, aiming at cleaning some part of our code with some refactoring.

File: src/extension/extension.ts [hunk 1]
```diff
@@ -437,7 +445,7 @@ export async function readExtensions(
           join(extensionsDirectory, extensionDir.name),
           extensionDir.name,
         );
-        if (ownedExtensions) {
+        if (ownedExtensions.length > 0) {
```
Comment (cscheid) [inline src/extension/extension.ts:448]: This change is ok, but is it necessary?

```
$ deno
Deno 1.41.0
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> let x = []
undefined
> let y = [1,2,3]
undefined
> if (x) { console.log("here"); y.push(...x) }
here
3
> y
[ 1, 2, 3 ]
```

Comment (cderv) [reply src/extension/extension.ts:448]: It is just that 
````
if (ownedExtensions)
````
will always be true... so I thought it was made for undefined, but it never is. That is what I added the length condition. 

But indeed `push(...[])` is ok. So maybe no if is needed there... 

unless it is there for other reason and I need to learn something new.

File: src/extension/install.ts [hunk 2]
```diff
@@ -339,7 +339,7 @@ async function readAndCopyExtensions(
   const extensions = await readExtensions(extensionsDir);
   info(
     `    Found ${extensions.length} ${
-      extensions.length === 1 ? "extension" : "extensions"
+      extensions.length <= 1 ? "extension" : "extensions"
```
[the remark this answers is not in the document]
Comment (cderv) [reply src/extension/install.ts:342]: "0 extensions" => 🤯 

I didn't know that ! Thanks !

**Pull request merged**